mongo/client.go

```diff
@@ -384,6 +384,12 @@ func (c *Client) configure(opts *options.ClientOptions) error {
 			}
 			return ""
 		}),
+		topology.WithSRVMaxHosts(func(srvMaxHosts int) int {
```

Comment:
In order to access `SRVMaxHosts` during SRV polling, we use the same strategy as with `SRVServiceName`.